src/main/perl/LV.pm

```diff
@@ -98,6 +118,62 @@ sub new_from_system
 
 =pod
 
+=head2 lvcache_ks
+
+Print the ks code for the cache on the logical volume.
+
+=cut
+
+sub lvcache_ks
+{
+    my $self = shift;
+    $self->{cache_lv}->create_ks;
+    my $command = join(" ", LVCONVERT, LVTYPE, 'cache', LVCACHEPOOL,
+       "$self->{volume_group}->{devname}/$self->{cache_lv}->{devname}", "$self->{volume_group}->{devname}/$self->{devname}");
+
+    print <<EOC 
+    if ! lvm lvs $self->{volume_group}->{devname}/$self->{devname} | awk '{ print \$5 }' | grep $self->{cache_lv}->{devname} > /dev/null
+    then
+        $command
+    fi
+EOC
+
+}
+
+
+=pod
+
+=head2 create_cache
+
+Creates the cache on the logical volume. Returns $? (0 if
+success). If the logical volume cache already exists, it returns 0 without
+doing anything.
+
+Returns 0 on success.
+
+=cut
+
+sub create_cache
+{
+    my $self = shift;
+    $self->{cache_lv}->create == 0 or return $?; 
+    my $output = CAF::Process->new([LVLVS, "$self->{volume_group}->{devname}/$self->{devname}"])->output();
+    my @lines = split(/\n/, $output);
+    if ($lines[1] && $lines[1] =~ /\s\[$self->{cache_lv}->{devname}\]\s/m){
+        $this_app->debug(5, "Cache $self->{cache_lv}->{devname} on logical volume $self->devpath already exists. Leaving"); 
+        return 0;
+    }
+    my $command = [LVCONVERT, LVTYPE, 'cache', LVCACHEPOOL, 
```

Comment:
if this is the same commandline as  n the `ks` method, maybe put it in a small private method that generates it, to be sure they are always equal